python_modules/dagster/dagster/_serdes/serdes.py

```diff
@@ -362,47 +450,70 @@ def __init__(
 
     def unpack(
         self,
-        storage_dict: Dict[str, Any],
+        unpacked_dict: Dict[str, UnpackedValue],
         whitelist_map: WhitelistMap,
-        descent_path: str,
+        context: UnpackContext,
     ) -> T_NamedTuple:
         try:
-            storage_dict = self.before_unpack(**storage_dict)
+            unpacked_dict = self.before_construct(context, unpacked_dict)
             unpacked: Dict[str, PackableValue] = {}
-            for key, value in storage_dict.items():
+            for key, value in unpacked_dict.items():
                 loaded_name = self.get_loaded_field_name(field=key)
                 # Naively implements backwards compatibility by filtering arguments that aren't present in
                 # the constructor. If a property is present in the serialized object, but doesn't exist in
                 # the version of the class loaded into memory, that property will be completely ignored.
                 if loaded_name in self.constructor_param_names:
-                    unpack_fn = self.get_field_unpack_fn(field=loaded_name)
-                    unpacked[loaded_name] = unpack_fn(
-                        value, whitelist_map=whitelist_map, descent_path=descent_path
-                    )
+                    # custom unpack regardless of hook vs recursive descent
+                    custom = self.field_serializers.get(loaded_name)
```

Comment:
I'm not following the reason for the change/expansion in complexity here-- why doesn't resolving one "unpack" function work and then performing a single assignment with its result work?

Reply:
this code path is very hot, `get_field_unpack_fn` was showing up as material in profiler results so i inlined it

Reply:
I did some additional inlining in this vein above and was able to drop cpu further 
```
Target 0 size 826.7KiB:
mean time from 10 samples: 17.147ms

Target 1 size 4.2KiB:
mean time from 10 samples: 0.149ms

Target 2 size 3.6KiB:
mean time from 10 samples: 0.139ms

Target 3 size 8.8KiB:
mean time from 10 samples: 0.195ms

Target 4 size 1.8MiB:
mean time from 10 samples: 47.278ms

Target 5 size 23.8MiB:
mean time from 10 samples: 474.157ms
```

Reply:
Interesting, ok